backend/proxy-server/src/server.js

```diff
@@ -44,65 +48,83 @@ app.all('/*', upload.none(), (req, res, next) => {
   const clientIp = req.connection.remoteAddress;
   const allowlist = conduit.allowlist;
   if (allowlist.length > 0 && clientIp !== '127.0.0.1') {
-    let allowed = false,
-      inactive = 0;
+    let allowedIP = null;
     for (let i = 0, imax = allowlist.length; i < imax; i++) {
       const item = allowlist[i];
       if (item.status === 'active' && item.ip === clientIp) {
-        allowed = true;
+        allowedIP = item;
         break;
       }
-
-      if (item.status === 'inactive') {
-        inactive++;
-      }
+    }
+    if (!allowedIP) {
+      return next(
+        new RestApiError(403, {
+          client: `${clientIp} restricted`,
+        })
+      );
     }
 
-    if (!allowed && inactive !== allowlist.length) {
-      // console.log('---->', clientIp, allowlist, inactive);
-      return next(new RestApiError(403, { client: `${clientIp} restricted` }));
+    if (allowedIP.status !== 'active') {
```

Comment:
Instead of counting for inactive IPs, I checked for if the current allowed ip status is active